### DISCUSSION TITLE
Best way to save file based on template?

### DISCUSSION THREAD

**Discussion by fredrikaverpil** [epwalsh/obsidian.nvim#499] [Q&A] [answered]
Hi 👋 and thanks for this very fine project!

I would like to understand how to work with templates. I can create a new document based on a template with `:ObsidianTemplate`, but this doesn't automatically define a filepath (like `:ObsidianNew` does) so I can't just hit `:w` to save the document. Instead I have to actually define the full path to my vault and then the filename I want.

I also cannot run `:ObsidianRename` as this assumes I've already saved the file first.

What's a good workflow here?
I am looking to set up templates: project, resource, meeting notes etc.

**Comment by epwalsh** [comment, marked as answer]
Hey @fredrikaverpil, sounds like we might need a new command for this, but you could also do this programmatically with obsidian.nvim's Lua API. For example:

```lua
local client = require("obsidian").get_client()
client:create_note { title = "Foo", dir = client.dir / "notes" / "project1", template = "project_template" }
```

**Reply by fredrikaverpil** [reply]
Hey @epwalsh and thanks for the suggestion.

I tried this, and for some reason, the frontmatter from the template isn't used. For example, I have `date: "{{date}}"` in the `project_template` frontmatter, but that doesn't go into the new note.

```yml
---
id: project template
aliases:
  - "{{title}}"
tags: 
date: "{{date}}"
---
```

If I run `:ObsidianTemplate project_template` the date is correctly inserted.

Do you see this on your end too?

I'm also getting an error unless I add `.md` to the end of the template in the lua command (the actual file on disk resides in `Templates/project_template.md` in my vault and I've configured the plugin to use the "Templates" folder.

```lua
local client = require("obsidian").get_client()
client:create_note { title = "Foo", dir = client.dir / "projects", template = "project_template.md" }
client:open_note(note)
```

**Reply by epwalsh** [reply]
@fredrikaverpil seems to work fine for me. Make sure you have the latest commit.

**Reply by fredrikaverpil** [reply]
I must have been on an old commit or something, yes. It works fine now with your original code. Thanks!
